geode-protobuf/src/test/java/org/apache/geode/internal/protocol/AuthenticationIntegrationTest.java

```diff
@@ -143,4 +229,111 @@ public void simpleAuthenticationSucceeds() throws Exception {
+
+    AuthenticationAPI.SimpleAuthenticationResponse authenticationResponse =
+        parseSimpleAuthenticationResponseFromInput();
+    assertFalse(authenticationResponse.getAuthenticated());
+  }
+
+  @Test
+  public void simpleAuthenticationWithInvalidCreds() throws Exception {
+    cache = createCacheWithSecurityManagerTakingExpectedCreds();
+    setupCacheServerAndSocket();
+
+    ClientProtocol.Message authenticationRequest = ClientProtocol.Message.newBuilder()
+        .setRequest(ClientProtocol.Request.newBuilder()
+            .setSimpleAuthenticationRequest(AuthenticationAPI.SimpleAuthenticationRequest
+                .newBuilder().putCredentials(ResourceConstants.USER_NAME, TEST_USERNAME)
+                .putCredentials(ResourceConstants.PASSWORD, "wrong password")))
+        .build();
+
+    authenticationRequest.writeDelimitedTo(outputStream);
+
+    AuthenticationAPI.SimpleAuthenticationResponse authenticationResponse =
+        parseSimpleAuthenticationResponseFromInput();
+    assertFalse(authenticationResponse.getAuthenticated());
+  }
+
+  @Test
+  public void noAuthenticatorSet() throws IOException {
+    cache = createNoSecurityCache();
+    setupCacheServerAndSocket();
+
+    // expect the cache to be in what we recognize as a no-security state.
+    assertFalse(((InternalCache) cache).getSecurityService().isIntegratedSecurity());
+    assertFalse(((InternalCache) cache).getSecurityService().isClientSecurityRequired());
+    assertFalse(((InternalCache) cache).getSecurityService().isPeerSecurityRequired());
+  }
+
+  @Test
+  public void someAuthenticatorSet() throws IOException {
+    createLegacyAuthCache("security-client-authenticator");
+    setupCacheServerAndSocket();
+
+    ClientProtocol.Message authenticationRequest = ClientProtocol.Message.newBuilder()
+        .setRequest(ClientProtocol.Request.newBuilder().setSimpleAuthenticationRequest(
+            AuthenticationAPI.SimpleAuthenticationRequest.newBuilder()))
+        .build();
+
+    authenticationRequest.writeDelimitedTo(outputStream);
+
+    AuthenticationAPI.SimpleAuthenticationResponse authenticationResponse =
+        parseSimpleAuthenticationResponseFromInput();
+    assertFalse(authenticationResponse.getAuthenticated());
+  }
+
+  @Test
+  public void peerAuthenticatorSet() throws IOException {
+    createLegacyAuthCache("security-peer-authenticator");
+    setupCacheServerAndSocket();
+
+    ClientProtocol.Message authenticationRequest = ClientProtocol.Message.newBuilder()
+        .setRequest(ClientProtocol.Request.newBuilder().setSimpleAuthenticationRequest(
+            AuthenticationAPI.SimpleAuthenticationRequest.newBuilder()))
+        .build();
+
+    authenticationRequest.writeDelimitedTo(outputStream);
+
+    AuthenticationAPI.SimpleAuthenticationResponse authenticationResponse =
+        parseSimpleAuthenticationResponseFromInput();
+    assertFalse(authenticationResponse.getAuthenticated());
```

Comment:
We really should expect an `ErrorResponse` here rather than the generic `AuthenticationResponse`.

Reply:
Hmm, surprised this wasn't failing prior as the legacy service wasn't return a proper message.   Fixed now.

Reply:
Fun, turns out this was working because of the issue we found where we'll get a stale security manager from a previous test, now that I fixed it the test is actually broken until I rebase from develop.